### PR TITLE
Add action and related CI

### PR DESCRIPTION
**Pull request by mwestphal**
Actual usage can be seen here: https://github.com/f3d-app/f3d-superbuild/pull/192

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+inputs:
+  key:
+    description: 'Cache key to differentiate between caches'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Install sccache Windows
```
Comment (Meakk) [inline action.yml:15]: What about Linux? It's already installed?

Comment (mwestphal) [reply action.yml:15]: yes, and see https://github.com/f3d-app/sccache-setup-action/issues/5

**Comment by mwestphal** [comment]
Require ninja on Windows: https://github.com/mozilla/sccache/issues/957

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        path: 'source'
+        fetch-depth: 0
+
+    - name: Checkout HelloWorld
+      uses: actions/checkout@v4
+      with:
+        repository: cristianadam/HelloWorld
```
Comment (Meakk) [inline .github/workflows/ci.yml]: we do not have control over this repo, can we use something else?

Comment (mwestphal) [reply .github/workflows/ci.yml]: we can use anything that has no dependency, do you have a suggestion ?

This repository is maintained and has releases, maybe I could point a specific release to checkout instead ?

Comment (Meakk) [reply .github/workflows/ci.yml]: I'm just concerned about the repository being closed.  
But yes, at least point to a tag.

**Review by Meakk — APPROVED** [review]

**Pull request merged**